File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/items/instances/WoodenEgg.java
package it.paranoidsquirrels.idleguildmaster.storage.data.items.instances;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.Utils;
import it.paranoidsquirrels.idleguildmaster.storage.data.pets.Pet;

/* JADX INFO: loaded from: classes3.dex */
public class WoodenEgg extends it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Egg {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Consumable, it.paranoidsquirrels.idleguildmaster.storage.data.items.Item
    protected void configureProperties() {
        this.idName = R.string.egg_wooden_name;
        this.idDescription = R.string.egg_wooden_description;
        this.idImage = R.drawable.egg_wooden;
        this.source.add(Integer.valueOf(R.string.dungeon_name_enchanted_forest));
        this.source.add(Integer.valueOf(R.string.dungeon_name_the_southern_grove));
        this.price = 100L;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Egg
    public Pet hatch() {
        int iCalculateNewPetId = Utils.calculateNewPetId();
        double dRandom = Utils.random();
        if (dRandom < 0.75d) {
            return Pet.getInstance("FloatingSeed", iCalculateNewPetId);
        }
        return dRandom < 0.95d ? Pet.getInstance("WalkingBush", iCalculateNewPetId) : Pet.getInstance("HolyTree", iCalculateNewPetId);
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/items/instances/WoodenEgg.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.items.instances

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.Utils
import it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Egg
import it.paranoidsquirrels.idleguildmaster.storage.data.pets.Pet

class WoodenEgg : Egg() {
    override fun configureProperties() {
        idName = R.string.egg_wooden_name
        idDescription = R.string.egg_wooden_description
        idImage = R.drawable.egg_wooden
        source.add(R.string.dungeon_name_enchanted_forest)
        source.add(R.string.dungeon_name_the_southern_grove)
        price = 100L
    }

    override fun hatch(): Pet? {
        val id = Utils.calculateNewPetId()
        val r = Utils.random()
        return if (r < 0.75) {
            Pet.getInstance("FloatingSeed", id)
        } else if (r < 0.95) {
            Pet.getInstance("WalkingBush", id)
        } else {
            Pet.getInstance("HolyTree", id)
        }
    }
}
